### ISSUE TITLE
Adding JCenter to resolvers doesn't work

### ISSUE THREAD

**Issue by dwijnand** [sbt/sbt#2465]
## steps

``` scala
resolvers += Resolver.jcenterRepo
```
## problem

`externalResolvers` doesn't contain JCenter:

``` scala
> show externalResolvers
[info] ArrayBuffer(
  FileRepository(local,FileConfiguration(true,None),Patterns(ivyPatterns=List(${ivy.home}/local/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), artifactPatterns=List(${ivy.home}/local/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), isMavenCompatible=false, descriptorOptional=false, skipConsistencyCheck=false)),
  public: https://repo1.maven.org/maven2/,
  public: https://repo1.maven.org/maven2/
)
```
## expectation

`externalResolvers` shouldn't remove from `resolvers`
## notes
- https://github.com/sbt/sbt/blob/v0.13.10/ivy/src/main/scala/sbt/Resolver.scala#L247-L249
- #2335

**Comment by eed3si9n** [comment]
The problem is that the JCenter can slip in from our launcher's repository list. So removing JCenter is intentional design unless your solution can distinguish between what's been added by the launcher vs human.

**Comment by dwijnand** [comment]
I'd be happy with a solution that does that distinction. Do you know how to do that?

**Comment by eed3si9n** [comment]
https://github.com/sbt/sbt/blob/0.13.10/main/src/main/scala/sbt/Defaults.scala#L1148-L1153:

``` scala
    externalResolvers <<= (externalResolvers.task.?, resolvers, appResolvers, useJCenter) {
      case (Some(delegated), Seq(), _, _) => delegated
      case (_, rs, Some(ars), uj)         => task { ars ++ Resolver.reorganizeAppResolvers(rs, uj, true) }
      case (_, rs, _, uj)                 => task { Resolver.withDefaultResolvers(rs, uj) }
    },
    appResolvers <<= appConfiguration apply appRepositories,
```

Doesn't `appResolvers` come from the launcher configuration? I think we need to filter things there.

**Comment by eed3si9n** [comment]
Given that the method I added is called `reorganizeAppResolvers` maybe I was thinking about filtering things from `ars`, but somehow I messed up or forgot, it seems.

**Comment by dwijnand** [comment]
Solved in #2467
